src/SeleniumLibrary/keywords/browsermanagement.py

```diff
@@ -117,6 +117,70 @@ def open_browser(self, url, browser='firefox', alias=None,
         uses. Notice that prior to SeleniumLibrary 3.0, the library
         contained its own profile that was used by default.
 
+        Optional ``options`` argument allows to define browser specific
+        Selenium options. Example for Chrome, the ``options`` argument
+        allows defining the following
+        [https://seleniumhq.github.io/selenium/docs/api/py/webdriver_chrome/selenium.webdriver.chrome.options.html#selenium.webdriver.chrome.options.Options|methods and attributes]
+        and for Firefox these
+        [https://seleniumhq.github.io/selenium/docs/api/py/webdriver_firefox/selenium.webdriver.firefox.options.html?highlight=firefox#selenium.webdriver.firefox.options.Options|methods and attributes]
+        are available. Please note that not all browsers supported by the
+        SeleniumLibrary have Selenium options available, please consult
+        the Selenium documentation which browser do support the Selenium
+        options. Selenium options are also supported, when ``remote_url``
+        argument is used.
+
+        The SeleniumLibrary ``options`` argument accepts Selenium
+        options in three different formats 1) As a sting, 2) List containing
+        dictionaries 3) Python object which is an instance of the
+        Selenium options object.
+
+        The sting format is allows to define Selenium options methods
+        or attributes and it's arguments in string format. Methods and
+        attributes are separated by a comma and arguments are separated
+        with a colon. Example like this:
+        `argument:arg1,other_argument:arg2:arg3,attribute:value`.
+
+        The method and attributes are case and space sensitive
+        and must match to the Selenium options methods and attributes names.
+        It is possible to have space before and after the comma for readability.
+        Example `argument:arg1,argument:arg2` and
+        `argument:arg1 , argument:arg2` are converted to same
+        object internally. Spaces around arguments are not removed.
+        Arguments are always stings and are not converted to other types.
+        If there is need to define a literal colon for an argument,
+        it can be escaped by using literal backslash. Please
+        note that backslash is an escape character in Robot Framework and
+        therefore backslash must escaped. Example:
+        `argument:arg1_with\\\\:colon`.
+
+        List containing dictionaries format is similar to the earlier
+        string format, but defines the methods and attributes and their
+        arguments using Robot Framework lists and dictionaries.
+        Method or attribute arguments, must be defined in a list. Method
+        or argument must be a dictionary key and argument list must be
+        the dictionary value. Dictionary must contain only one key
+        value pair. The dictionary must be placed in a list. Example:
+        `[{"argument": ["arg1"]}, {"other_argument": ["arg1", "arg2"]}]`
```

Comment:
This format looks pretty strange. Why not to have just one dictionary that has all possible attribute/value pairs? For me it seems `[{'x': 1}, {'y': 2}]` and `{'x': 1, 'y': 2}` have same information but the latter format is a lot simpler especially in Robot's data.

Reply:
There are some methods in Selenium options that takes two arguments in, example like https://selenium-python.readthedocs.io/api.html#selenium.webdriver.firefox.options.Options.set_capability and presenting two arguments as a list was an easiest option.

I know that it's an overkill when a method or an attribute has a single argument, but I initially didn't want to support two formats. I initially thought that it would be harder to explain in the documentation and thought that it would bring complicity to the code. But when I am now looking at the matter I am not anymore so sure. The code will not be a that much of a problem but what would be a good way to explain it?

Reply:
Also many methods can be called many times, like the https://selenium-python.readthedocs.io/api.html#selenium.webdriver.firefox.options.Options.add_argument and that was not possible in the dictionary only format.

Reply:
Explained in bad way, please ignore the above comments. Lets try again. 

The dictionary only format is not possible because many Selenium options methods are called (usually) many times. 
Example `{"add_argument": "value1", "add_argument": "value2"}` is not possible in the dictionary format. Therefore there is list in the outermost definition: `[{...}, {...}]`. 

Also there are Selenium options methods which have two arguments and therefore there is a list inside of the dictionary. But those methods are actually a minority and based on the found internet examples, not used that much. Therefore supporting two different formats could be be possible: 1) If the only one argument, list inside of the dictionary would be optional, like: `[{"add_argument": "value1"}, {"add_argument": "value2"}]` 2) If there two (or more) arguments, list inside of the dictionary is mandatory, like `[{"set_capability": ["key1", "value1"]}, {"set_capability": ["key2", "value2"]]`. I did initially reject the idea of two formats, because because it would be harder to explain the documentation and would bring complexity to the code. But now when looking at the matter, code will not be complex, only one `if else` statement, but will it be easy to explain in the documentation.

Also in some point, I was thinking not the support the `[{...}, {...}]` format, but Selenium contains some stupid code. [like in headless](https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/chrome/options.py#L151) where Python `True` is explicitly required. It was also possible to make automatic conversion in the string format, but I did not want to go in that path, because it was hard to decide what format should be converted.

I agree that the  `[{...}, {...}]` format is messy and not easy to create in RF data. Would you see other and better options?

Reply:
1. Does `[{'method': 'x'}, {'method': 'y'}]` map to single call with two arguments (`obj.method('x', 'y')`) or to two individual calls with one argument (`obj.method('x'); obj.method('y')`)?

2. Instead of a list with one item dictionaries the format could be a list with internal lists (or tuples). Then `[('method', 'x'), ('method', 'y', 'z')]` could be interpreted as `obj.method('x'); obj.method('y', 'z')`. This ought to be pretty generic. It would handle even cases where arguments needs to be a list such as `[('method', [1, 2, 3])]` that would be `obj.method([1, 2, 3])`.

3. If the common case is setting attributes or calling methods only once and with one argument, then it might be a good idea to support specifying methods/attributes as a dictionary like `{'method': 'arg', 'attribute': 'value'}`. In the code this could be trivially converted to a list of lists (technically tuples) with code like `if isinstance(item, Mapping): item = item.items()`.

Reply:
Is there a way in the string format to call a method with more than one argument? Also, it's somewhat annoying if the API needing actual `True` makes it impossible to use this format.

What if the string format would be like `method('arg'); attr = 'value'; method('foo', True)`? This would be somewhat technical, but it would also look exactly like the same Selenium code in Python. It would require strings to be quotes but I don't think that's a huge problem. Parsing wouldn't be trivial but shouldn't be overly complicated either. I can help if needed.

Hmm, if this string format works well enough, then perhaps there wouldn't be need for the list of dicts/lists format at all. Externally created option objects probably should still be supported.

Reply:
I agree that the `method('arg'); attr = 'value'; method('foo', True)` format is better. Although there is some more work to write it and also parsing is not trivial the benefits are greater than those two downsides. 

1) It is clear for the reader is user calling an attribute or a method.
2) There is a clear separation between Python booleans (Like `True`), numbers (although I haven't found an example where number would be needed) and strings.
3) This replaces the hard to make and read `[{...}, {...}]` format. 
4) Documentation will hopefully be less confusing, because there is only one format to explain. 

Yes external objects should be supported, but that is trivial.